src/serialize.h

```diff
@@ -1020,6 +1052,34 @@ void Unserialize(Stream& is, std::list<T, A>& l, int nType, int nVersion)
 }
 
 
+/**
+ * Collections w/ iterators (vector, list, dequeue...) with type casting.
+ * For example, this is useful in combination with smart pointers for polymorphism.
+ */
+
+template<typename Stream, typename T, typename C, typename D>
+void Serialize(Stream& os, const C& l, int nType, int nVersion)
+{
+    WriteCompactSize(os, l.size());
+    for (typename C::const_iterator it = l.begin(); it != l.end(); ++it) {
+        Serialize<Stream, T, D>(os, (*it), nType, nVersion);
+    }
+}
+
+template<typename Stream, typename T, typename C, typename D>
+std::void_t<typename C::value_type> // enforce that C has member value_type, as it happens for STL collections
+Unserialize(Stream& is, C& l, int nType, int nVersion)
+{
+    l.clear();
+    unsigned int nSize = ReadCompactSize(is);
+    typename C::iterator it = l.begin();
```

Comment:
Is this needed ?

Reply:
No it indeed was a leftover, nice catch!